Makefile

```diff
@@ -42,16 +42,19 @@ licenses: ## Verifies dependency licenses and requires GITHUB_TOKEN to be set
 	go build $(GOFLAGS) -o karpenter cmd/controller/main.go
 	golicense hack/license-config.hcl karpenter
 
+## Assumes developer is working with AWS. Remove line 52 if you are not working with AWS.
```

Comment:
I'd remove this comment. It's not really tenable to ask users to modify makefiles locally. We will need a better solution for other clouds, let's solve it when we get there. Further, this doesn't actually break anything if you're not on AWS, it just applies a bogus annotation that does nothing. 

In the long run, we might want to do something like check if CLOUD_PROVIDER is AWS and do something special, but let's hold off for now.